pkg/relayer/session/sessiontree.go

```diff
@@ -206,3 +208,18 @@ func (st *sessionTree) Delete() error {
 	// Delete the KVStore from disk
 	return os.RemoveAll(filepath.Dir(st.storePath))
 }
+
+// ClaimOnce adds the sessionTree to the list of sessionTrees to be claimed.
+// It returns the updated list of sessionTrees or the original list if the sessionTree
+// is already included in a list for sessions to be claimed.
+func (st *sessionTree) ClaimOnce(sessionTreesToClaim []relayer.SessionTree) []relayer.SessionTree {
```

Comment:
This is a very confusing function for me. 

An object receives a list of objects and returns that list while potentially (or not) mutating its own state. 

Doesn't feel like it follows SOLID principles and I just struggle to follow along.

Reply:
I see what you mean.

The constraints here are:
* `sessionMu` is private to the `SessionTree`
* `isClaiming` has to be altered only once
* Append the the slice only when `isClaiming` has changed

I'll try another approach but open to any suggestion.

Reply:
Updated, PTAL

Reply:
See new comment in latest review.